File: Facultate/Anul 2/Sem 2/MPP/Laburi/CSharpLab_3/Shared/network/Response.cs

using System.Text.Json.Serialization;

namespace Shared.network;

public class Response
{
    [JsonInclude]
    public ResponseType Type { get; private set; }

    [JsonInclude]
    public object? Data { get; private set; }

    [JsonInclude]
    public string? Message { get; private set; }

    public Response(ResponseType type, object? data, string? message)
    {
        Type = type;
        Data = data;
        Message = message;
    }

    public static Response Error(string message)
    {
        return new Response(ResponseType.Error, null, message);
    }
}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/JavaTemaORM/Shared/src/main/java/network/Response.java
package network;

import java.io.Serializable;

public class Response implements Serializable {
    private final ResponseType type;
    private final Object data;
    private final String message;

    public Response(ResponseType type, Object data, String message) {
        this.type = type;
        this.data = data;
        this.message = message;
    }

    public static Response ok(Object data) {
        return new Response(ResponseType.OK, data, null);
    }

    public static Response error(String message) {
        return new Response(ResponseType.ERROR, null, message);
    }

    public ResponseType getType() {
        return type;
    }

    public Object getData() {
        return data;
    }

    public String getMessage() {
        return message;
    }
}
